tests/views/user_preference_api_tests.py

```diff
@@ -0,0 +1,181 @@
+import json
+import os, uuid
+from django.test import TransactionTestCase
+from django.urls import reverse
+from django.test.utils import captured_stdout
+from django.core import management
+from arches.app.models.models import UserPreference
+from arches.app.models.system_settings import settings
+from arches.app.models import models
+from tests import test_settings
+from django.test.client import Client
+
+
+# these tests can be run from the command line via
+# python manage.py test tests.views.user_preference_api_tests --settings="tests.test_settings"
+
+
+class UserPrefApiTest(TransactionTestCase):
+
+    serialized_rollback = True
+
+    def generate_user_preference(self):
+        user_preference = UserPreference()
+        user_preference.userpreferenceid = uuid.uuid4()
+        user_preference.user = models.User.objects.get(username="admin")
+        user_preference.preferencename = "test preference"
+        user_preference.config = [
+            {"overlayid": "7d0dffba-5bcf-4694-a5d7-3425ad97fa2b", "sortorder": 1},
+            {"overlayid": "85c0cdd4-95d0-4350-bd9d-729f326fe1d5", "sortorder": 2},
+        ]
+        return user_preference
+
+    def user_preference_json_data(self, username):
+        user_pref = {
+            "userpreferenceid": None,
+            "user": username,
+            "preferencename": "test preference",
+            "config": [
+                {"overlayid": "7d0dffba-5bcf-4694-a5d7-3425ad97fa2b", "sortorder": 1},
+                {"overlayid": "85c0cdd4-95d0-4350-bd9d-729f326fe1d5", "sortorder": 2},
+            ],
+        }
+        return user_pref
+
+    def test_user_preference_api(self):
```

Comment:
Thanks for providing such thorough tests. Kind of a shame we don't have a framework to delegate more of this to. I can take a closer look on the next iteration once some of these error conditions go away, but you might consider breaking some of these out into discrete tests.